packages/react/src/__tests__/ReactElementClone-test.js

```diff
@@ -274,8 +274,55 @@ describe('ReactElementClone', () => {
 
     const root = ReactDOMClient.createRoot(document.createElement('div'));
     await act(() => root.render(<Grandparent />));
-    expect(component.childRef).toEqual({current: null});
-    expect(component.parentRef.current.xyzRef.current.tagName).toBe('SPAN');
+    if (gate(flags => flags.enableRefAsProp && flags.disableStringRefs)) {
+      expect(component.childRef).toEqual({current: null});
+      expect(component.parentRef.current.xyzRef.current.tagName).toBe('SPAN');
+    } else if (
+      gate(flags => !flags.enableRefAsProp && !flags.disableStringRefs)
+    ) {
+      expect(component.childRef).toEqual({current: null});
+      expect(component.parentRef.current.xyzRef.current.tagName).toBe('SPAN');
+    } else if (
+      gate(flags => flags.enableRefAsProp && !flags.disableStringRefs)
+    ) {
+      expect(component.childRef).toEqual({current: null});
+      expect(component.parentRef.current.xyzRef.current.tagName).toBe('SPAN');
+    } else {
+      // Not going to bother testing every possible combination.
+    }
+  });
+
+  it('should steal the ref if a new string ref is specified', async () => {
+    if (gate(flags => !flags.enableRefAsProp && !flags.disableStringRefs)) {
+      await expect(async () => {
+        // create an element without an owner
+        const element = React.createElement('div', {id: 'some-id'});
+        class Parent extends React.Component {
+          render() {
+            return <Child>{element}</Child>;
+          }
+        }
+        let child;
+        class Child extends React.Component {
+          render() {
+            child = this;
+            const clone = React.cloneElement(this.props.children, {
+              ref: 'xyz',
+            });
+            return <div>{clone}</div>;
+          }
+        }
+
+        const root = ReactDOMClient.createRoot(document.createElement('div'));
+        await act(() => root.render(<Parent />));
+        expect(child.refs.xyz.tagName).toBe('DIV');
+      }).toErrorDev([
```

Comment:
before the fix this failed, hitting the case where there was no owner during string ref conversion